datagrid_gtk3/utils/transformations.py

```diff
@@ -448,5 +449,9 @@ def degree_decimal_str_transform(value, length=8):
     if len(value) < length:
         value = '0' * (length - len(value)) + value
 
+    # Add decimal point at the expected location
     value = '{}.{}'.format(value[:2], value[2:])
-    return float(value)
+
+    # Remove non-significant leading zeroes
+    value = re.sub(r'0+(\d\.)', r'\1', value)
```

Comment:
I wasn't happy with regular expressions nor with using `float`, so your suggestion about `decimal.Decimal` turns out to be what I was looking for. Thanks.